algorithms/src/polycommit/sonic_pc/mod.rs

```diff
@@ -340,10 +346,10 @@ impl<E: PairingEngine, S: AlgebraicSponge<E::Fq, 2>> SonicKZG10<E, S> {
                 proof
             });
         }
-        let batch_proof = pool.execute_all().into_iter().collect::<Result<_, _>>().map(BatchProof).map_err(Into::into);
+        let batch_proof = pool.execute_all().into_iter().collect::<Result<_, _>>().map(BatchProof)?;
```

Comment:
Dramatic savings of three characters. Probably applies to other places too.
```suggestion
        let batch_proof = pool.execute_all().into_iter().collect::<Result<_>>().map(BatchProof)?;
```